wf-galleon/EAP7-1217_Galleon_User_FP_Builder.adoc

```diff
@@ -0,0 +1,192 @@
+= EAP7-1217: Tooling for end user creation of Galleon feature packs
+:author:            Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Offer the ability to package WildFly server customization content as Galleon feature-pack. 
+These feature-packs would then be installed on top of WildFly server Galleon installation. 
+One of the main use cases is cloud image building:
+
+* Internally, when building our own image, custom content could have been packaged as feature-packs and installed.
+
+* Externally, inside an OpenShift S2I context. User would have the ability to build (during S2I assembly or offline)
+feature-packs containing piece of custom configuration and custom JBoss Module modules (eg: JDBC drivers).
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1217[EAP7-1217]
+
+* https://issues.jboss.org/browse/WFGP-130[WFGP-130]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-891[EAP7-891]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* OpenShift
+
+=== Other Interested Projects
+
+* WildFly Galleon
+
+== Requirements
+
+In order to address this requirement, we are defining a new WildFly Maven plugin (_org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin_)
+goal named _build-user-feature-pack_.
+This goal handles part of the complexity one would be faced when building a feature-pack for WildFly. 
+With minimal configuration, a user should be able to package his content into a WildFly compliant feature-pack.
+
+=== Possible usages inside an S2I context
```

Comment:
One thing is to think about how to handle backwards compatibility, namely how will applications using current S2I mechanisms act with this new configuration possibilities.

Reply:
Yes, the s2i workflow should work the same if Galleon is not used.